src/cbrkit/sim/collections.py

```diff
@@ -90,11 +90,15 @@ def __call__(self, x: Sequence[V], y: Sequence[V]) -> float:
     pass
 
 try:
+    from dataclasses import dataclass
+    from typing import Collection, Callable, Any, List, Tuple, Union
     import numpy as np
 
+
     @dataclass(slots=True)
-    class dtw[V](SimPairFunc[Collection[V], float]):
-        """Dynamic Time Warping similarity function.
+    class dtw:
```

Comment:
The generics should not be removed, this comment also applies to all other instances in this PR where they were removed. I just pushed some changes using proper types everywhere, please rebase your changes on the last revision.